tooling/ssa_fuzzer/fuzzer/src/mutations/mod.rs

```diff
@@ -1,36 +1,90 @@
 mod commands_mutator;
 mod configuration;
+mod function;
 mod instructions;
 mod witness_mutator;
 
-use crate::fuzz_lib::fuzz_target_lib::FuzzerData;
+use crate::fuzz_lib::function_context::FunctionData;
 use crate::mutations::configuration::{
-    BASIC_MUTATION_CONFIGURATION, MAX_NUMBER_OF_MUTATIONS, MutationOptions,
+    BASIC_FUNCTION_VEC_MUTATION_CONFIGURATION, FunctionVecMutationOptions,
 };
+use libfuzzer_sys::arbitrary::Unstructured;
 use rand::{Rng, rngs::StdRng};
 
-pub(crate) fn mutate(data: FuzzerData, rng: &mut StdRng) -> FuzzerData {
-    let (mut blocks, mut commands, mut initial_witness) =
-        (data.blocks, data.commands, data.initial_witness);
-    let number_of_mutations = rng.gen_range(1..MAX_NUMBER_OF_MUTATIONS);
-    for _ in 0..number_of_mutations {
-        match BASIC_MUTATION_CONFIGURATION.select(rng) {
-            MutationOptions::InstructionBlocks => {
-                instructions::mutate_vec_instruction_block(&mut blocks, rng);
-            }
-            MutationOptions::FuzzerCommands => {
-                commands_mutator::mutate_vec_fuzzer_command(&mut commands, rng);
-            }
-            MutationOptions::Witnesses => {
-                let index = rng.gen_range(0..initial_witness.len());
-                witness_mutator::witness_mutate(&mut initial_witness[index], rng);
-            }
+trait MutateFunctionVec {
+    fn mutate(rng: &mut StdRng, funcs: &mut Vec<FunctionData>);
+}
+struct RandomMutation;
+impl MutateFunctionVec for RandomMutation {
+    fn mutate(rng: &mut StdRng, functions: &mut Vec<FunctionData>) {
+        let mut bytes = [0u8; 128];
+        rng.fill(&mut bytes);
+        *functions = Unstructured::new(&bytes).arbitrary().unwrap();
```

Comment:
You are replacing all the functions with random functions. If I understand how this works, this basically removes the usefulness of everything done before